engine/src/main/java/com/datatorrent/stram/ResourceRequestHandler.java

```diff
@@ -108,6 +113,23 @@ public void updateNodeReports(List<NodeReport> nodeReports)
     }
   }
 
+  public List<String> getNodesExceptHost(List<String> hostNames)
```

Comment:
We assign a single host for each request. However, Yarn ContainerRequest can have list of nodes.. Added hostnames to set for contains check.